front/components/assistant/conversation/MessageItem.tsx

```diff
@@ -90,21 +94,61 @@ const MessageItem = React.forwardRef<HTMLDivElement, MessageItemProps>(
         }
       );
 
-    if (message.visibility === "deleted") {
-      return null;
-    }
-
     const messageFeedbackWithSubmit: FeedbackSelectorProps = {
       feedback: messageFeedback
         ? {
             thumb: messageFeedback.thumbDirection,
             feedbackContent: messageFeedback.content,
+            isConversationShared: messageFeedback.isConversationShared,
           }
         : null,
       onSubmitThumb,
       isSubmittingThumb,
     };
 
+    const router = useRouter();
+    const [hasScrolledToMessage, setHasScrolledToMessage] = useState(false);
+    const [highlighted, setHighlighted] = useState(false);
+    // Because the prop ref can be undefined
+    const scrollRef = React.useRef<HTMLDivElement>(null);
+
+    // Effect: scroll to the message and temporarily highlight if it is the anchor's target
+    useEffect(() => {
+      if (!router.asPath.includes("#")) {
+        return;
+      }
+      const messageIdToScrollTo = router.asPath.split("#")[1];
+      if (
+        messageIdToScrollTo === sId &&
+        !hasScrolledToMessage &&
+        (ref || scrollRef)
+      ) {
+        setTimeout(() => {
+          setHasScrolledToMessage(true);
+          // Use ref to scroll to the message
+          const divRef = ref
+            ? (ref as React.RefObject<HTMLDivElement>)
+            : scrollRef;
+          if (divRef.current) {
+            divRef.current.scrollIntoView({
+              behavior: "smooth",
+              block: "center",
+            });
+          }
+          setHighlighted(true);
+
+          // Highlight the message for a short time
+          setTimeout(() => {
+            setHighlighted(false);
+          }, 2000);
```

Comment:
I presume this is not the most desired way to do this no? This should be handled by CSS? I know we have prior art in sparkle. Can you check with @JulesBelveze or @Duncid on the best way to do that?